packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.cpp

```diff
@@ -362,20 +362,25 @@ namespace Teuchos {
           // Compute the set intersection / union as specified by the
           // enum.
           Array<string> newNames;
-          if (setOp == Intersection)
-            std::set_intersection (globalNames.begin(), globalNames.end(),
-                                   otherNames.begin(), otherNames.end(),
-                                   std::back_inserter (newNames));
-          else if (setOp == Union)
-            std::set_union (globalNames.begin(), globalNames.end(),
-                            otherNames.begin(), otherNames.end(),
-                            std::back_inserter (newNames));
-          else
-            TEUCHOS_TEST_FOR_EXCEPTION(setOp != Intersection && setOp != Union,
-                               std::logic_error,
-                               "Invalid set operation enum value.  Please "
-                               "report this bug to the Teuchos developers.");
-          globalNames.swap (newNames);
+          if ( std::is_sorted(globalNames.begin(), globalNames.end()) &&
```

Comment:
The only issue with this, is that if both sets of names just happen to be sorted, then this will sort, instead of just leaving new names at the end like you want.  You should really just add an option rather than trying to guess what the user wants.

Reply:
If both names are sorted, that is fine to have the new list sorted.  Really, what can't happen is a reordering of things within the global list.